features/collection/thresholding/impl/src/main/java/org/opennms/netmgt/threshd/ThresholdEvaluatorHighLow.java

```diff
@@ -201,32 +211,53 @@ public Status evaluateAfterFetch(double dsValue) {
             return Status.NO_CHANGE;
         }
 
-        protected boolean isThresholdExceeded(double dsValue) {
+        protected boolean isThresholdExceeded(double dsValue, Double value) {
```

Comment:
Since this method and isRearmExceeded will NPE if given nulls can we update the method signature to take a primitive. That way their preconditions will be enforced.

Reply:
Doesn't look like this changed, anyways not a big deal...

Reply:
Oh, yeah. method with only one argument is only used in tests so added `@VisibleForTesting` annotation and other will always pre-check for null so this seems safe.

Reply:
Not that it matters at this point, but the reason I originally commented on this is because the method would assert its preconditions by using a primitive `double` in the signature rather than implicitly only being valid with a non-null `Double`.

If we had changed the other code to return `OptionalDouble` rather than a nullable `Double` we would have either `empty` or a primitive `double` to pass in and we would satisfy the preconditions.

No need to action this, just explaining my train of thought.